### PR TITLE
Add a create-function script and document it

### PR DESCRIPTION
**Pull request by schlueter**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,21 +5,11 @@ This Lambda function consumes SQS messages with information necessary to populat
...
-        --role arn:aws:iam::327361568963:role/lambda-sqs
-
-This will setup the initial lambda function. The SQS queue it will consume from must be created separately and (to my current knowledge) manually added as a trigger to the function.
-
-Update the parameters as appropriate.
+Use the scripts:
+
+    REDROUTE_REDIS_HOST=some-redis-host REDROUTE_REDIS_PORT=56379 ./create-function.sh $(./build-and-deploy.sh some-nonexistant-prefix) the-prefix-you-want
+
+With the correct prerequisites, this will create a Lambda function, and an SQS queue, each named *the-prefix-you-want-redroute* and a Lambda event source mapping between the two.
```
Comment (johnnyshankman) [inline README.md]: This checks out and aligns with the idea presented here:

https://github.com/alleyinteractive/refinery29/blob/production/mu-plugins/refinery29-main/inc/routes/class-route-client.php#L45-L48

Which means we should be able to make the `$this->queue_url` dynamic, in the sense that we can assume what the URL of the SQS queue *should* be and write a handler for when that guess leads to a 404/error.

Hardcoded Config: https://github.com/alleyinteractive/refinery29/blob/production/mu-plugins/refinery29-main/inc/routes/class-route-client.php#L61

Where the dynamic assign could happen instead:
https://github.com/alleyinteractive/refinery29/blob/production/mu-plugins/refinery29-main/inc/routes/class-route-client.php#L101

File: README.md [hunk 2]
```diff
@@ -5,21 +5,11 @@ This Lambda function consumes SQS messages with information necessary to populat
...
-        --role arn:aws:iam::327361568963:role/lambda-sqs
-
-This will setup the initial lambda function. The SQS queue it will consume from must be created separately and (to my current knowledge) manually added as a trigger to the function.
-
-Update the parameters as appropriate.
+Use the scripts:
+
+    REDROUTE_REDIS_HOST=some-redis-host REDROUTE_REDIS_PORT=56379 ./create-function.sh $(./build-and-deploy.sh) some-prefix
+
+With the correct prerequisites, this will create a Lambda function, and an SQS queue, each named *some-prefix-redroute* and a Lambda event source mapping between the two.
```
Comment (johnnyshankman) [inline README.md]: This checks out and aligns with the idea presented here:

https://github.com/alleyinteractive/refinery29/blob/production/mu-plugins/refinery29-main/inc/routes/class-route-client.php#L45-L48

Which means we should be able to make the `$this->queue_url` dynamic, in the sense that we can assume what the URL of the SQS queue *should* be and write a handler for when that guess leads to a 404/error.

Hardcoded `queue` property that we shouldn't need anymore: https://github.com/alleyinteractive/refinery29/blob/production/mu-plugins/refinery29-main/inc/routes/class-route-client.php#L61

Where the dynamic assign could happen instead:
https://github.com/alleyinteractive/refinery29/blob/production/mu-plugins/refinery29-main/inc/routes/class-route-client.php#L101

**Comment by schlueter** [comment]
This specifically is not bash (bourne again shell), just sh (bourne shell), but your point stands.

To test, a user with appropriate access to aws may run

    REDROUTE_REDIS_HOST=ip-10-200-39-27.ec2.internal REDROUTE_REDIS_PORT=56379 ./create-function.sh $(./build-and-deploy.sh) some-test-name

Which will create an sqs queue named some-test-name-redroute, and a lambda function with the same name, and connect the two.

To test, in the sqs web console, a message like 

    {"command": "hset",
     "key": "some-key",
     "field": "some-field",
     "value": "some-value"}

may be sent to the create test queue, and then the result inspected on either cache05 or any of it's replicas or passthroughs. *Note that that machine is our production master redis server*, but if you're not testing in production, what's the point?

File: index.js [hunk 3]
```diff
@@ -6,7 +6,24 @@ const redis = require('redis');
...
+        // a individual error
+        return new Error('The server refused the connection');
+      }
+      if (options.error) {
+        return new Error(options.error);
+      }
+      if (options.total_retry_time > 1000 * 7) {
+        // End reconnecting after a specific timeout and flush all commands
+        // with a individual error
+        return new Error('Retry time exhausted');
```
Comment (johnnyshankman) [inline index.js]: I think in all these error cases you should actually just pass the error into callback, that way the lambda function exits correctly upon Error.

```
callback(new Error('The server refused the connection'));
...
callback(new Error(options.error));
...
callback(new Error('Retry time exhausted'));
```

Comment (johnnyshankman) [reply index.js]: That is, assuming that we want to stop retrying and exit the lambda function in these cases (that's what it looks like).

Comment (schlueter) [reply index.js]: I'll remove our update this tomorrow

Comment (schlueter) [reply index.js]: I was thinking this would trigger the error event which already used the callback

**Pull request merged**